src/app/static/src/js/components/runReport/runReport.vue

```diff
@@ -197,20 +212,33 @@
                 //TODO: Include parameters and changelog message when implemented
                 //TODO: Add link to running report log on response, when implemented
 
+                /*
+                 * This ensures parameter values are filled before running a report
+                 * if parameter values already pass validation, it will not re-validate
+                 * Also checks if parameter has values before validating
+                 */
+                if (!this.isValidParam && this.parameterValues.length) {
+                    this.validateParams()
+                    if (!this.isValidParam) {
+                        return
+                    }
+                }
+
                 //Orderly server currently only accepts a single instance value, although the metadata endpoint supports
                 //multiple instances - until multiple are accepted, send the selected instance value for instance with
                 //greatest number of options. See VIMC-4561.
                 let instances = {};
                 if (this.metadata.instances_supported && this.metadata.instances &&
                     Object.keys(this.metadata.instances).length > 0) {
                     const instanceName = Object.keys(this.metadata.instances).sort((a, b) => this.metadata.instances[b].length - this.metadata.instances[a].length)[0];
-                    const instance = this.selectedInstances[instanceName];
-                    instances = Object.keys(this.metadata.instances).reduce((a, e) => ({[e]: instance, ...a}), {});
+                    const instance = this.selectedInstances[instanceName];instances = Object.keys(this.metadata.instances).reduce((a, e) => ({[e]: instance, ...a}), {});
                 }
-
+                const params = {
+                    parameterValues: this.parameterValues
+                };
                 api.post(`/report/${this.selectedReport}/actions/run/`, {
                     instances: instances,
-                    params: {}, //TODO mrc-2167
+                    params: params,
```

Comment:
This params object you've made will add an extra layer of nesting to the posted `params` value that we don't want. Just `params: this.parameterValues` should be fine.